src/components/TextInput/TextInput.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { shallow } from 'enzyme';
+import React from 'react';
+
+import { Icon } from '../Icon/Icon';
+
+import { TextInput } from './TextInput';
+
+describe('<TextInput />', () => {
+	it('Should be able to render', () => {
+		shallow(<TextInput />);
+	});
+
+	it('Should set the correct className', () => {
+		const inputComponent = shallow(<TextInput />);
+
+		const inputElement = inputComponent.find('input');
+
+		expect(inputElement.hasClass('c-input')).toEqual(true);
+	});
+
+	it('Should pass on the id', () => {
+		const id = 'test';
+
+		const inputComponent = shallow(<TextInput id={id} />);
+
+		const inputElement = inputComponent.find('input');
+
+		expect(inputElement.prop('id')).toEqual(id);
+	});
+
+	it('Should be able to set the disabled state', () => {
+		const inputComponent = shallow(<TextInput disabled />);
+
+		const inputElement = inputComponent.find('input');
+
+		expect(inputElement.prop('disabled')).toEqual(true);
+	});
+
+	it('Should be able to set the placeholder value', () => {
+		const placeholder = 'this is a test placeholder';
+
+		const inputComponent = shallow(<TextInput placeholder={placeholder} />);
+
+		const inputElement = inputComponent.find('input');
+
+		expect(inputElement.prop('placeholder')).toEqual(placeholder);
+	});
+
+	it('Should be able to set an initial value', () => {
+		const defaultValue = 'default test value';
+
+		const inputComponent = shallow(<TextInput defaultValue={defaultValue} />);
+
+		const inputElement = inputComponent.find('input');
+
+		expect(inputElement.prop('value')).toEqual(defaultValue);
+	});
+
+	it('Should be able to render with an icon', () => {
```

Comment:
add test for keyboard navigation + check outline styles are present

> Extra form guidelines (http://master.viaa2.mono.digital/styleguide/docs/forms.html)
> The label needs to be tied to the form field (use the for="" attribute and an id for accessibility reasons)
> Forms must be keyboard accessible
> Test your forms using the keyboard
> Never remove focus outline states in CSS

Reply:
De ids worden correct doorgegeven van de props, dat zit in de testen. Checken of de keyboardnavigatie daar bijgevolg dus voor werkt is een feature van de browser zelf, en dus volgens mij niet te testen door ons.